src/stream/src/executor/sync_kv_log_store.rs

```diff
@@ -272,111 +305,157 @@ impl<S: StateStore> SyncedKvLogStoreExecutor<S> {
+                        drop(write_future_state);
                         let (stream, mut write_state, either) = result?;
                         match either {
                             WriteFutureEvent::UpstreamMessageReceived(msg) => {
                                 match msg {
                                     Message::Barrier(barrier) => {
-                                        if let Some(mutation) = barrier.mutation.as_deref() {
-                                            match mutation {
-                                                Mutation::Pause => {
-                                                    pause_stream = true;
-                                                }
-                                                Mutation::Resume => {
-                                                    pause_stream = false;
-                                                }
-                                                _ => {}
-                                            }
-                                        }
-
-                                        let write_state_post_write_barrier = Self::write_barrier(
-                                            &mut write_state,
-                                            barrier.clone(),
-                                            &self.metrics,
-                                            truncation_offset,
-                                            &mut buffer,
-                                        )
-                                        .await?;
-                                        seq_id = FIRST_SEQ_ID;
-                                        let update_vnode_bitmap =
-                                            barrier.as_update_vnode_bitmap(self.actor_context.id);
-                                        let barrier_epoch = barrier.epoch;
-                                        yield Message::Barrier(barrier);
-                                        write_state_post_write_barrier
-                                            .post_yield_barrier(update_vnode_bitmap.clone())
-                                            .await?;
-                                        if let Some(vnode_bitmap) = update_vnode_bitmap {
-                                            // Apply Vnode Update
-                                            read_state.update_vnode_bitmap(vnode_bitmap);
-                                            initial_write_epoch = barrier_epoch;
-                                            input = stream;
-                                            initial_write_state = write_state;
-                                            continue 'recreate_consume_stream;
-                                        } else {
-                                            write_future = WriteFuture::receive_from_upstream(
+                                        if clean_state
+                                            && buffer.no_flushed_items()
```

Comment:
I think the condition should be buffer.has_unflushed_items()? Otherwise, even for an empty buffer we will enter the unclean state.

Reply:
Fixed, since only one other area uses this, I just add a new method call: `is_empty`.

Then the predicate is `!buffer.is_empty() && buffer.no_flushed_items()`.

Calling `buffer.has_unflushed_items` seems ambiguous, because it implies having unflushed items, but doesn't indicate that there should be no flushed items.

Reply:
FYI, the implementation at the moment has more than one usages on `no_flushed_items`.

Reply:
The make things simple, I think we can just pause when previously in clean state && the barrier is checkpoint barrier && buffer is non-empty. It means simply avoid checkpoint barrier skip rows with some efforts.

Reply:
Fixed, thanks